src/main/scala/io/kf/etl/common/Utils.scala

```diff
@@ -0,0 +1,19 @@
+package io.kf.etl.common
+
+object Utils {
+
+    def calculateDataCategory(
+                               availableDataType: Seq[String],
+                               mapOfDataCategory_ExistingTypes: Map[String, Seq[String]]
+                             ): Set[String] = {
+      val cleanAvailableDataTypes = availableDataType.map(_.toLowerCase.trim)
+      mapOfDataCategory_ExistingTypes.collect { case t if isIncludedInAvailableDt(t._2, cleanAvailableDataTypes) => t }.keySet
```

Comment:
I think you can make this simplier with { case (dataCategories, availableDataTypes) if  cleanAvailableDataTypes.exists(t => dataCategories.contains(t))

check https://stackoverflow.com/questions/37888614/how-to-check-if-collection-contains-any-element-from-other-collection-in-scala for an example